### PR TITLE
fix(@aws-amplify/analytics): Fix attributes and userAttributes case transform on updateEndpoint.

### PR DESCRIPTION
**Pull request by temideoye**
Fixed a bug that transformed attributes and userAttributes object keys to lowercase while processing updateEndpoint requests.

*Issue:*
`_updateEndpoint()` transforms event object keys (such as eventId, name, session) to lowercase before sending to `_endpointRequest()` for request formatting. In the process, *attributes* and *userAttributes* object keys are inadvertently transformed.

Essentially, the whitelist variable supplied to the transform function is filtering out *Attributes* and *UserAttributes* which does not exist on the event object. Instead, it should filter out *attributes* and *userAttributes*.

*Description of changes:*
Changed values of the whitelist variable to `['attributes', 'userAttributes']`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by powerful23** [comment]
@temideoye the function is mainly for the backward compatibility. It works as expected. Do you have any issue caused by this?

**Comment by temideoye** [comment]
> @temideoye the function is mainly for the backward compatibility. It works as expected. Do you have any issue caused by this?

@powerful23, indeed. Please see screenshot from Pinpoint console below. Because updateEndpoint was transforming my attribute keys to lowercase, Pinpoint created new attributes instead of writing values to existing attributes.

**User Attributes**
attachment 1

**Custom Attributes**
attachment 2

File: packages/analytics/src/Providers/AWSPinpointProvider.ts [hunk 1]
```diff
@@ -364,7 +364,7 @@ export default class AWSPinpointProvider implements AnalyticsProvider {
         
         const request = this._endpointRequest(
             config, 
-            JS.transferKeyToLowerCase(event, [], ['Attributes', 'UserAttributes'])
+            JS.transferKeyToLowerCase(event, [], ['attributes', 'userAttributes'])
```
Comment (powerful23) [inline packages/analytics/src/Providers/AWSPinpointProvider.ts]: Please change it to `JS.transferKeyToLowerCase(event, [], ['attributes', 'userAttributes', 'Attributes', 'UserAttributes'])`

Comment (temideoye) [reply packages/analytics/src/Providers/AWSPinpointProvider.ts]: Of course, @powerful23. I've updated the commit.

**Pull request merged**